src/navigation/_CollapsingNavigationMenu.jsx

```diff
@@ -0,0 +1,162 @@
+   */
+  static openKeysToItem(menuItems, selectedPath) {
+    return CollapsingNavigationMenu.keysToItem(menuItems, selectedPath).reduce((acc, path) => {
+      acc[path] = true;
+      return acc;
+    }, {});
+  }
+
+  static getDerivedStateFromProps({ menuItems, selectedPath }, state) {
+    const newState = {};
+    if (state.previousSelectedPath !== selectedPath) {
+      newState.openKeys = Object.assign({}, state.openKeys, CollapsingNavigationMenu.openKeysToItem(menuItems[0], selectedPath));
+      newState.previousSelectedPath = selectedPath;
+    }
+    return newState;
+  }
+
+  constructor(props) {
+    super(props);
+    const { menuItems, selectedPath } = props;
+    this.renderMenuItems = this.renderMenuItems.bind(this);
+    this.handleOnClick = this.handleOnClick.bind(this);
+    this.handleKeyDown = this.handleKeyDown.bind(this);
+
+    this.state = {
+      previousSelectedPath: selectedPath,
+      openKeys: CollapsingNavigationMenu.openKeysToItem(menuItems[0], selectedPath),
+    };
+  }
+
+  handleKeyDown(event, item) {
+    if (event.nativeEvent.keyCode === KeyCode.KEY_SPACE || event.nativeEvent.keyCode === KeyCode.KEY_RETURN) {
+      this.handleOnClick(event, item);
+    }
+  }
+
+  handleOnClick(event, item) {
+    const { onSelect } = this.props;
+    const { openKeys } = this.state;
+
+    if (!item.childItems) {
+      onSelect(item.path);
+      return;
+    }
+    openKeys[item.path] = !openKeys[item.path];
+    this.setState({ openKeys });
+  }
+
+  renderMenuItems(menuItems, firstLevel) {
+    const { selectedPath } = this.props;
+    const { openKeys } = this.state;
+
+    if (!menuItems) {
+      return undefined;
+    }
+
+    return menuItems.map((item) => {
+      const itemIsOpen = openKeys[item.path];
+      const itemHasChildren = item.childItems !== undefined;
+
+      return (
+        <React.Fragment key={item.path}>
+          <div className={!firstLevel ? cx('indent') : null}>
+            <div
+              className={cx(['item', { 'is-selected': selectedPath === item.path }])}
+              tabIndex="0"
+              role="link"
+              aria-haspopup={itemHasChildren}
+              onKeyDown={event => this.handleKeyDown(event, item)}
+              onClick={event => this.handleOnClick(event, item)}
+            >
+              {itemHasChildren ? <span className={cx('disclosure')}>{ itemIsOpen ? <IconCaretDown className={cx('caret')} /> : <IconCaretRight className={cx('caret')} />}</span> : null}
+              {item.name}
+            </div>
+            {itemIsOpen ? this.renderMenuItems(item.childItems) : null}
+          </div>
+        </React.Fragment>
+      );
+    });
+  }
+
+  render() {
+    const { menuItems } = this.props;
+    return (
+      <div className={cx('collapsing-navigation-menu')} id="terra-dev-site-nav-menu" tabIndex="-1">
+        {menuItems ? this.renderMenuItems(menuItems[0].childItems, true) : undefined}
```

Comment:
Could there ever be multiple menu items? I see we keep grabbing the first array item

Reply:
no, there will only be one passed in